internal/exec/util/passwd.go

```diff
@@ -111,18 +110,10 @@ func (u Util) EnsureUser(c types.PasswdUser) error {
 
 // CheckIfUserExists will return Info log when user is empty
 func (u Util) CheckIfUserExists(c types.PasswdUser) (bool, error) {
-	code := -1
-	cmd := exec.Command(distro.ChrootCmd(), u.DestDir, distro.IdCmd(), c.Name)
-	stdout, err := cmd.CombinedOutput()
-	if err != nil {
-		if exitErr, ok := err.(*exec.ExitError); ok {
-			code = exitErr.Sys().(syscall.WaitStatus).ExitStatus()
-		}
-		if code == 1 {
-			u.Info("checking if user \"%s\" exists: %s", c.Name, fmt.Errorf("[Attention] %v: Cmd: %s Stdout: %s", err, log.QuotedCmd(cmd), stdout))
-			return false, nil
-		}
-		u.Logger.Info("error encountered (%T): %v", err, err)
+	usr, err := u.userLookup(c.Name)
+	if _, ok := err.(user.UnknownUserError); ok {
+		return false, nil
+	} else if err != nil || usr == nil {
```

Comment:
Nit: I think that here both the `else` and the `|| usr == nil` are non-idiomatic. The latter in particular I think would belong to the previous logic case (user not found).